core/src/smartcontracts/isi/asset.rs

```diff
@@ -103,6 +103,45 @@ pub mod isi {
         }
     }
 
+    impl Execute for Transfer<Asset, Metadata, Account> {
+        #[metrics(+"transfer_store")]
+        fn execute(self, _authority: &AccountId, wsv: &mut WorldStateView) -> Result<(), Error> {
+            let asset_id = self.source_id;
+            assert_asset_type(&asset_id.definition_id, wsv, AssetValueType::Store)?;
+
+            let (original_owner_metadata, destination_id) = {
+                let original_store_owner: &Account = wsv.account(&asset_id.account_id)?;
+                let metadata: Metadata = original_store_owner.metadata().clone();
+                let destination_id =
+                    AssetId::new(asset_id.definition_id.clone(), self.destination_id.clone());
+                (metadata, destination_id)
+            };
+
+            let asset_metadata_limits = wsv.config.asset_metadata_limits;
+            let destination_store_asset =
+                wsv.asset_or_insert(destination_id.clone(), Metadata::new())?;
+            let store: &mut Metadata = destination_store_asset
+                .try_as_mut()
+                .map_err(eyre::Error::from)
+                .map_err(|e| Error::Conversion(e.to_string()))?;
+
+            for pair in original_owner_metadata.iter() {
+                let key: &Name = pair.0;
+                let value: &Value = pair.1;
+                store.insert_with_limits(key.clone(), value.clone(), asset_metadata_limits)?;
+            }
+
+            {
+                let original_store_owner = wsv.account_mut(&asset_id.account_id)?;
+                assert!(original_store_owner.remove_asset(&asset_id).is_some());
+            }
+
+            wsv.emit_events([AssetEvent::Deleted(asset_id)]);
```

Comment:
There should be `Added` event as well

Reply:
I think in this case `Created` would be better because `Added` will expect `amount` parameter

Reply:
Yeah, sounds nice, let's do